pkg/applicationserver/registry.go

```diff
@@ -33,6 +34,109 @@ type DeviceRegistry interface {
 	Set(ctx context.Context, ids ttnpb.EndDeviceIdentifiers, paths []string, f func(*ttnpb.EndDevice) (*ttnpb.EndDevice, []string, error)) (*ttnpb.EndDevice, error)
 }
 
+type deprecatedDeviceField struct {
```

Comment:
Lets extract all of this into an internal package instead of directly copy-pasting from NS like this.
My suggestion is:
`pkg/internal/registry` with at least following names exported:
- `DeprecatedDeviceField`
- `DeprecatedDeviceFieldMatch`
- `MatchDeprecatedDeviceFields`

Ideally I think we should also extract utilities like `WrapDeviceGetWithDeprecatedFields` etc., but that can be done as second step.